src/inc/til/winrt.h

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    namespace details
+    {
+        template<typename T>
+        constexpr T winrt_empty_value() noexcept
+        {
+            if constexpr (std::is_base_of_v<winrt::Windows::Foundation::IUnknown, T>)
+            {
+                return nullptr;
+            }
+            else
+            {
+                return {};
+            }
+        }
+    };
+
+    template<typename T>
+    struct property
+    {
+        property<T>(const T& defaultValue = details::winrt_empty_value<T>()) :
+            _value{ defaultValue } {}
+
+        property<T>& operator=(const property<T>& other) = default;

```

Comment:
You don't need to explicitly write `<T>` inside the template. You do need to declare the other 3 standard operators and constructors though, or otherwise the AuditMode complains. This shows btw, that our AuditMode isn't working for TerminalControl once again. 😅